openeo/rest/job.py

```diff
@@ -351,14 +357,17 @@ def __repr__(self):
             n=self.name, t=self.metadata.get("type", "unknown"), h=self.href
         )
 
-    def download(self, target: Optional[Union[Path, str]] = None, chunk_size=None) -> Path:
+    def download(self, target: Optional[Union[Path, str]] = None, chunk_size: Optional[int] = None) -> Path:
```

Comment:
I would change the default here directly:
```suggestion
    def download(self, target: Optional[Union[Path, str]] = None, chunk_size: int = DEFAULT_DOWNLOAD_CHUNK_SIZE) -> Path:
```

That simplifies implementation and is self-documenting.
And otherwise the user can't intentionally disable chunking anymore if they want that for some reason